src/Sylius/Bundle/CoreBundle/Resources/config/app/sylius.yml

```diff
@@ -191,6 +191,7 @@ sylius_promotion:
         promotion_coupon:
             classes:
                 model: Sylius\Component\Core\Model\Coupon
+                repository: Sylius\Bundle\PromotionBundle\Doctrine\ORM\CouponRepository
```

Comment:
1. This should be configured in `CoreBundle`, but in `PromotionBundle`
2. The class should be placed in `.../doctrine/orm/orm.xml` file